crates/rover-client/src/query/config/is_federated.rs

```diff
@@ -17,33 +17,33 @@ use graphql_client::*;
 /// Snake case of this name is the mod name. i.e. publish_partial_schema_mutation
 pub struct IsFederatedGraph;
 
-#[derive(Debug, PartialEq)]
-pub struct IsFederatedGraphResponse {
-    pub result: bool,
-}
-
 pub fn run(
     variables: is_federated_graph::Variables,
     client: &StudioClient,
-) -> Result<IsFederatedGraphResponse, RoverClientError> {
+) -> Result<(), RoverClientError> {
+    let graph = variables.graph_id.clone();
     let data = client.post::<IsFederatedGraph>(variables)?;
-    let is_federated_response = data.service.unwrap();
-    Ok(build_response(is_federated_response))
+    build_response(data, graph)
 }
 
-type FederatedResponse = is_federated_graph::IsFederatedGraphService;
 type ImplementingServices = is_federated_graph::IsFederatedGraphServiceImplementingServices;
 
-fn build_response(service: FederatedResponse) -> IsFederatedGraphResponse {
+fn build_response(
+    data: is_federated_graph::ResponseData,
+    graph: String,
+) -> Result<(), RoverClientError> {
+    let service = data.service.ok_or(RoverClientError::NoService {
+        graph: graph.clone(),
+    })?;
     match service.implementing_services {
         Some(typename) => match typename {
-            ImplementingServices::FederatedImplementingServices => {
-                IsFederatedGraphResponse { result: true }
-            }
+            ImplementingServices::FederatedImplementingServices => Ok(()),
             ImplementingServices::NonFederatedImplementingService => {
-                IsFederatedGraphResponse { result: false }
+                Err(RoverClientError::ExpectedFederatedGraph { graph })
```

Comment:
It does not quite make sense to me that a non-federated graph would warrant a `RoverClientError`. It's only a non-federated graph, which itself is not an error. This doesn't work in the particular context of Rover trying to publish a non-federated graph to a federated endpoint, but it might work in other contexts. So I think if there needs to be an error, it should be formulated where Rover uses this check. 

That being said, I don't think it makes sense for this to have a whole response type like it was before, and should really just return a result of a <bool, RoverClientError>, where RoverClientError takes care of a MalformedResponse, which is a legit ClientError in the case of `None`.

What do you think about this sort of approach?

Reply:
Mmmmmm. I think it makes sense for the `is_federated` query by itself to return a bool like this, but I think maybe we should consider moving where the request actually lives. Instead of making `is_federated` a public query, what if we nested that query inside of our `rover-client::subgraph::check` and return this error there if a `convert: bool` parameter is set to false.

My thinking here is that we likely want consumers of the library to be using the Studio API in the same way that we've outlined in Rover. The conversion from a non-federated graph to a federated graph is a one-way operation so we should likely be guarding against that in the library as well. 

What do you think?

Reply:
Ah! I am fine with having `is_federated` a `pub(crate)` method that gets used in `subgraph::check` and `subgraph::publish` and whichever future queries we do that needs this check. It totally makes sense to me that the `subgraph::publish` query should guard against publishing a non-federated graph to a federated graph.

Reply:
OK! This is done 😄